Automattic/jetpack
# Jetpack Start v2

Tools for partners to provision Jetpack sites.

Jetpack Start allows third parties to provide and pay for Jetpack plans via a secure API call, and shortcuts some setup steps (optionally).

This PR is just a starting point, all it really does is move the register() and authorize() steps into a wp-cli command. Eventually the endpoint will do all kinds of other setup and checks on the WP side so that users are given the plans that partners have signed up for, etc.

Usage:

```
CALYPSO_ENV=$OPTIONAL_CALYPSO_ENV JETPACK_START_API_HOST=$OPTIONAL_API_HOST sh $JETPACK_PLUGIN_PATH/bin/partner-provision.sh --partner_id=$PARTNER_ID --partner_secret=$PARTNER_SECRET --plan=$PLAN_SLUG --user_id=$WP_USER_ID
```

Testing instructions:

- Merge D5467-code
- Create an oAuth client and whitelist it in the JPPHP partner install 2.0 endpoint
- Run `JETPACK_START_API_HOST=your.sandbox.host.com sh ./bin/partner-provision.sh --partner_id=$CLIENT_ID --partner_secret=$CLIENT_SECRET --plan=anything --user_id=$WP_USER_ID` on a new WP install
- Grab the returned JSON, decode it. There should be a property called "next_url". Paste that into your browser to complete installation of Jetpack.

Right now, this doesn't do any setup beyond regular Jetpack Connect. Eventually it will. I want to get this merged before further work is done, since 99% of that work will be changes to WPCOM and this lets us test against a production version of Jetpack.
